src/Util/Profiler.php

```diff
@@ -85,6 +87,60 @@ public function __construct(Cache $configCache)
 		$this->reset();
 	}
 
+	/**
+	 * Start a profiler recording
+	 *
+	 * @param string $value
+	 * @return void
+	 */
+	public function startRecording(string $value)
+	{
+		if (!$this->enabled) {
+			return;
+		}
+
+		$this->timestamps[] = ['value' => $value, 'stamp' => microtime(true), 'credit' => 0];
+	}
+
+	/**
+	 * Stop a profiler recording
+	 *
+	 * @param string $callstack
+	 * @return void
+	 */
+	public function stopRecording(string $callstack = '')
+	{
+		if (!$this->enabled || empty($this->timestamps)) {
+			return;
+		}
+
+		$timestamp = array_pop($this->timestamps);
+
+		$duration = floatval(microtime(true) - $timestamp['stamp'] - $timestamp['credit']);
+		$value = $timestamp['value'];
+
+		foreach ($this->timestamps as $key => $stamp) {
+			$this->timestamps[$key]['credit'] += $duration;
+		}
+
+		$callstack = $callstack ?: System::callstack(4, $value == 'rendering' ? 0 : 1);
+
+		if (!isset($this->performance[$value])) {
+			// Prevent ugly E_NOTICE
```

Comment:
This comment is entirely unnecessary, initializing variables is fine on its own.
```suggestion
```